### PR TITLE
Feature/static api

### PR DESCRIPTION
**Pull request by Hadatko**
Hi @MichalPrincNXP ,
this is WIP PR. It is not best and it is touching other platforms/env. I was working in nxp directory structure so some cases, which are now different (some files are generic here, but in nxp repo they looked env/platform specific) need be fixed. Could you propose me suggestions for this PR? Thank you.

Related to #13

### CODE REVIEW

**Comment by MichalPrincNXP** [comment]
Hi Dusan, thank you, the proposal looks ok to me. I have couple of comments that will be added into the code review.

**Comment by MichalPrincNXP** [comment]
Also, I am afraid we could get into troubles when having multiinstance. I am writing it here as this needs to be analyzed.

**Comment by Hadatko** [comment]
> Also, I am afraid we could get into troubles when having multiinstance. I am writing it here as this needs to be analyzed.

Sure, from my point of you, it shouldn't be worse or better now in multiinstance case. I was trying to keep static context in same scope as semaphore, mutex,...

**Comment by MichalPrincNXP** [comment]
Dusan, as per [this instruction](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork), could you please allow me to edit this PR. I would like to pass the update from my side to be reviewed and tested by you? Do you agree? Thank you.

**Comment by Hadatko** [comment]
Sure, i agree, just i don't see simple way how can you push directly to this branch.

**Comment by Hadatko** [comment]
I cannot see the checkbox for allowing that. Maybe due to: https://github.com/isaacs/github/issues/1681

File: lib/include/environment/freertos/rpmsg_env_specific.h [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+#if (configSUPPORT_STATIC_ALLOCATION == 0)
+#warning You have configured RPMsg_Lite to use static API but FreeRTOS is not configured for static allocations! Please switch the configSUPPORT_STATIC_ALLOCATION to 1 in your FreeRTOSConfig.h file.
+#endif
+
+#include "FreeRTOS.h"
+
+typedef StaticSemaphore_t LOCK_STATIC_CONTEXT;
+typedef StaticQueue_t rpmsg_static_queue_ctxt;
+
+/* Queue object static storage size in bytes, should be defined as (RL_BUFFER_COUNT*sizeof(rpmsg_queue_rx_cb_data_t)) */
```
Comment (Hadatko) [inline lib/include/environment/freertos/rpmsg_env_specific.h:31]: I am concered about usage of rpmsg_queue_rx_cb_data_t without including header file with declaration.

File: lib/include/rpmsg_env.h [hunk 2]
```diff
@@ -155,7 +156,16 @@ void env_memcpy(void *dst, void const *src, uint32_t len);
 int32_t env_strcmp(const char *dst, const char *src);
 void env_strncpy(char *dest, const char *src, uint32_t len);
 int32_t env_strncmp(char *dest, const char *src, uint32_t len);
+#ifdef MCUXPRESSO_SDK
```
Comment (Hadatko) [inline lib/include/rpmsg_env.h:159]: For me personaly this looks like it doesn't belongs here. We moved OS specific things into its header file. 
This looks like same approach should be used.

File: lib/include/environment/freertos/rpmsg_env_specific.h [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+ *
+ * DESCRIPTION
+ *
+ *       This file contains FreeRTOS specific constructions.
+ *
+ **************************************************************************/
+#ifndef RPMSG_ENV_SPECIFIC_H_
+#define RPMSG_ENV_SPECIFIC_H_
+
+#if defined(RL_USE_STATIC_API) && (RL_USE_STATIC_API == 1)
```
Comment (Hadatko) [inline lib/include/environment/freertos/rpmsg_env_specific.h:22]: Also missing header file for definition of RL_USE_STATIC_API

Comment (MichalPrincNXP) [reply lib/include/environment/freertos/rpmsg_env_specific.h:22]: Agree, could you please add #include "rpmsg_default_config.h" into all rpmsg_env_specific.h files?

File: lib/include/environment/freertos/rpmsg_env_specific.h (hunk 1 above)
Comment (MichalPrincNXP) [reply lib/include/environment/freertos/rpmsg_env_specific.h:31]: How about to move the rpmsg_queue_rx_cb_data_t typedef from the rpmsg_queue.h directly into the rpmsg_env_specific.h files?

File: lib/include/rpmsg_env.h (hunk 2 above)
Comment (MichalPrincNXP) [reply lib/include/rpmsg_env.h:159]: I would keep it as is because this printf part is library-specific, not OS-specific.

**Review by MichalPrincNXP — CHANGES_REQUESTED** [review]
1. rpmsg_queue_rx_cb_data_t typedef placement needs to be finalized
2. #include "rpmsg_default_config.h" needs to be added into all rpmsg_env_specific.h

File: lib/include/environment/freertos/rpmsg_env_specific.h (hunk 3 above)
Comment (Hadatko) [reply lib/include/environment/freertos/rpmsg_env_specific.h:22]: I agree

File: lib/include/environment/freertos/rpmsg_env_specific.h (hunk 1 above)
Comment (Hadatko) [reply lib/include/environment/freertos/rpmsg_env_specific.h:31]: I agree

File: lib/include/rpmsg_env.h (hunk 2 above)
Comment (Hadatko) [reply lib/include/rpmsg_env.h:159]: ok

File: lib/include/rpmsg_env.h [hunk 4]
```diff
@@ -81,6 +82,7 @@
 #include <stdint.h>
 #include "rpmsg_default_config.h"
 #include "rpmsg_platform.h"
+#include "FreeRTOS.h"
```
[the remark this answers is not in the document]
Comment (Hadatko) [reply lib/include/rpmsg_env.h]: It looks like you were commenting very old commit

[the remark this answers is not in the document]
Comment (MichalPrincNXP) [reply lib/include/rpmsg_env.h]: odd, this comment comes from April timeframe, please ignore

File: lib/include/rpmsg_lite.h [hunk 5]
```diff
@@ -125,6 +126,9 @@ struct rpmsg_lite_instance
     struct virtqueue *tvq;              /*!< transmit virtqueue */
     struct llist *rl_endpoints;         /*!< linked list of endpoints */
     LOCK *lock;                         /*!< local RPMsg Lite mutex lock */
+#if defined(RL_USE_STATIC_API) && (RL_USE_STATIC_API == 1)
+    STACK lock_stack;                  /*!< Stack for lock */
+#endif
     uint32_t link_state;                /*!< state of the link, up/down*/
```
[the remark this answers is not in the document]
Comment (MichalPrincNXP) [reply lib/include/rpmsg_lite.h:132]: also the obsolete comment

File: lib/include/environment/freertos/rpmsg_env_specific.h (hunk 1 above)
Comment (MichalPrincNXP) [reply lib/include/environment/freertos/rpmsg_env_specific.h:33]: Will you that? Thanks.

Comment (Hadatko) [reply lib/include/environment/freertos/rpmsg_env_specific.h:41]: Done

**Comment by Hadatko** [comment]
Is it ok now?

**Comment by MichalPrincNXP** [comment]
> 
> 
> Is it ok now?

I think so, let me retest on my side once again, ok?

**Comment by Hadatko** [comment]
Sure.

**Review by MichalPrincNXP — APPROVED** [review]

**Comment by MichalPrincNXP** [comment]
Thank you for the effort, Dusan!

**Pull request merged**